test/Tests.Infrastructure/Extensions/ManualResetEventSlimExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Tests.Infrastructure.Extensions
+{
+    public static class ManualResetEventSlimExtensions
```

Comment:
Wouldn't be easier to use `AsyncManualResetEvent` instead?

Reply:
I followed the `WaitForEtlToComplete` and `WaitForEtl`.
I'd turn them into `async` with removal of `AsyncHelpers.RunSync` and return `AsyncManualResetEvent` as @arekpalinski mentioned, that has the `WaitAsync` capability already. 

I can imagine the number of awaits required as I probe the replication atm. I think it's a better way of structuring it though.

Reply:
I went with `AsyncManualResetEvent`. 
Getting rid from the `AsyncHelpers.RunSync` in those methods is a quest for a different time :-) I think...